pynumdiff/utils/evaluate.py

```diff
@@ -70,71 +70,68 @@ def plot_comparison(dt, dxdt_truth, dxdt_hat1, title1, dxdt_hat2, title2, dxdt_h
         axes[i].tick_params(axis='y', labelsize=15)
         axes[i].set_title(title, fontsize=18)
         if i==2: axes[i].legend(loc='lower right', fontsize=12)
-        _, _, rms_dxdt = rmse(np.zeros(dxdt_hat.shape), dt, np.zeros(dxdt_hat.shape), dxdt_hat, np.zeros(dxdt_hat.shape), dxdt_truth)
-        R_sqr = error_correlation(dxdt_hat, dxdt_truth)
-        axes[i].text(0.05, 0.95, f"RMSE = {rms_dxdt:.2f}\n$R^2$ = {R_sqr:.2g}",
+        rmse_dxdt = rmse(dxdt_truth, dxdt_hat)
+        R_sqr = error_correlation(dxdt_truth, dxdt_hat)
+        axes[i].text(0.05, 0.95, f"RMSE = {rmse_dxdt:.2f}\n$R^2$ = {R_sqr:.2g}",
                      transform=axes[i].transAxes, fontsize=15, verticalalignment='top')
 
     fig.tight_layout()
 
 
-def rmse(x, dt, x_hat, dxdt_hat, x_truth=None, dxdt_truth=None, padding=0):
-    """Evaluate x_hat based on RMSE, calculating different ones depending on whether :code:`dxdt_truth`
-    and :code:`x_truth` are known.
+def robust_rme(x, x_hat, padding=0, M=6):
```

Comment:
I decided to make this a separate function, although it reduces to be the same as `rmse` when M is infinity, because the two are used for different purposes in our optimization and plotting.